### ISSUE TITLE
SC2030 is triggered when the -v test is run using brackets

### ISSUE THREAD

**Issue by Crestwave** [koalaman/shellcheck#1409]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2030
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2086
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash

#!/usr/bin/env bash
var=1
([ -v var ]; :)
echo "$var"

```

#### Here's what shellcheck currently says:

```
Line 2:
([ -v var ]; :)
      ^-- SC2030: Modification of var is local (to subshell caused by (..) group).
 
Line 3:
echo "$var"
      ^-- SC2031: var was modified in a subshell. That change might be lost.
```

#### Here's what I wanted or expected to see:

No errors, as `-v` (introduced in bash 4.2) only tests if a variable is set and doesn't modify it as far as I know. Strangely enough, though, it behaves as expected when `test -v` is run explicitly.

**Comment by Gandalf-** [comment]
This appears to be intentional from a comment in the source
- https://github.com/koalaman/shellcheck/blob/master/src/ShellCheck/AnalyzerLib.hs#L487-L494

Introduced by this commit
- https://github.com/koalaman/shellcheck/commit/41b6e3d5eb81b18998dae6476901cfe60dbf6c7e

**Comment by koalaman** [comment]
The intention was a quick hack to avoid a `http_proxy is referenced but not assigned` in cases like this:

    if [ -v http_proxy ]
    then
      echo "Proxy set"
    fi

This subshell warning was an unintended side effect, and is definitely a bug. Nice catch!
